### PR TITLE
GH-14454: add link navigation to conflictsolver and the conflicted attribute

### LINKED ISSUE #14454: VC Configurator Conflict Solver Enhancement: Navigation
User should be able to navigate from a conflicting attribute to the respective conflict group and vice versa.

Wires: See https://4prsat.axshare.com/#id=bma4rm&p=d_cps_configuration_conflicts&g=1
https://4prsat.axshare.com/#id=fr0uxa&p=d_cps_configuration_forms&g=1

This affects ConfiguratorAttributeHeaderComponent and ConfiguratorGroupsService. The latter needs to be able to navigate to a conflict group for an attribute, and to a standard group, given the attribute from the conflict group. It needs to be injected into ConfiguratorAttributeHeaderComponent (breaking).
The attribute needs to display the navigation link in the conflict group case (also breaking).

Cover breaking changes by creating an issue for the migration schematics for the respective major release, and introduce deprecations and forVersion if needed (in case we release that with a minor)

### PR DESCRIPTION
**Pull request by Changsuwan**
The navigation, however, is not working properly yet

### CODE REVIEW

File: feature-libs/product-configurator/rulebased/components/attribute/header/configurator-attribute-header.component.ts [hunk 1]
```diff
@@ -137,4 +144,41 @@ export class ConfiguratorAttributeHeaderComponent
...
     return images && this.hasImage ? images[0] : undefined;
   }
+
+  navigateToGroup(): void {
+    this.configuratorCommonsService
+      .getConfiguration(this.owner)
+      .pipe(take(1))
+      .subscribe((configuration) => {
+        if (this.groupType === Configurator.GroupType.CONFLICT_GROUP) {
+          const groupId = this.attribute?.groupId;
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/attribute/header/configurator-attribute-header.component.ts]:
```suggestion
          const groupId = this.attribute.groupId;
```
attribute is always present

File: feature-libs/product-configurator/rulebased/components/attribute/header/configurator-attribute-header.component.ts [hunk 2]
```diff
@@ -137,4 +144,41 @@ export class ConfiguratorAttributeHeaderComponent
...
+          if (groupId) {
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/attribute/header/configurator-attribute-header.component.ts:158]: What if group is undefined here? Under which circumstances does that happen? (Not covered by test)

File: feature-libs/product-configurator/rulebased/components/attribute/header/configurator-attribute-header.component.ts [hunk 3]
```diff
@@ -137,4 +144,41 @@ export class ConfiguratorAttributeHeaderComponent
     const images = this.attribute.images;
     return images && this.hasImage ? images[0] : undefined;
   }
+
+  navigateToGroup(): void {
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/attribute/header/configurator-attribute-header.component.ts:151]: Please add JSDoc

File: feature-libs/product-configurator/rulebased/components/attribute/header/configurator-attribute-header.component.ts [hunk 4]
```diff
@@ -137,4 +144,41 @@ export class ConfiguratorAttributeHeaderComponent
...
+              configuration,
+              groupId
+            );
+            this.focusAttribute(this.attribute.name);
+          }
+        }
+      });
+  }
+
+  protected focusAttribute(name: string): void {
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/attribute/header/configurator-attribute-header.component.ts:177]: Please state that this is about the focus handling in the group that we are navigating to, so it does not affect the current view. Will make it easier to understand why we call the 'loading' method twice

File: feature-libs/product-configurator/rulebased/components/service/configurator-storefront-utils.service.ts [hunk 5]
```diff
@@ -138,6 +138,32 @@ export class ConfiguratorStorefrontUtilsService {
     }
   }
 
+  /**
+   * Find an attribute by its name in the form and focus it.
+   */
+  focusAttribute(name: string): void {
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/service/configurator-storefront-utils.service.ts:146]: 3 branches are not covered by tests for this method, can we improve the test coverage here?

File: feature-libs/product-configurator/rulebased/core/facade/configurator-groups.service.ts [hunk 6]
```diff
@@ -80,6 +80,27 @@ export class ConfiguratorGroupsService {
...
 
+  /**
+   * Navigates to the first non-conflict group of the configuration which is not completed.
+   * This method assumes that the configuration has incomplete groups,
+   * the caller has to verify this prior to calling this method. In case no incomplete group is
+   * present, nothing will happen
+   *
+   * @param {CommonConfigurator.Owner} owner - Configuration owner
+   */
+  navigateToFirstGroup(owner: CommonConfigurator.Owner): void {
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/core/facade/configurator-groups.service.ts]: Please remove. Not called at all

File: feature-libs/product-configurator/rulebased/core/facade/configurator-group-status.service.ts [hunk 7]
```diff
@@ -55,6 +55,21 @@ export class ConfiguratorGroupStatusService {
...
   }
 
+  /**
+   * Return the first group of a configuration or undefined
+   * if not present
+   *
+   * @param {Configurator.Configuration} configuration - Configuration
+   * @returns {Configurator.Group} Conflict group
+   */
+  getFirstGroup(
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/core/facade/configurator-group-status.service.ts]: Please remove. Not called at all

**Review by ChristophHi — CHANGES_REQUESTED** [review]

File: feature-libs/product-configurator/rulebased/components/attribute/header/configurator-attribute-header.component.ts (hunk 1 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/components/attribute/header/configurator-attribute-header.component.ts]: Agree

File: feature-libs/product-configurator/rulebased/components/attribute/header/configurator-attribute-header.component.ts (hunk 3 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/components/attribute/header/configurator-attribute-header.component.ts:151]: Done

File: feature-libs/product-configurator/rulebased/core/facade/configurator-group-status.service.ts (hunk 7 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/core/facade/configurator-group-status.service.ts]: Done!

File: feature-libs/product-configurator/rulebased/components/attribute/header/configurator-attribute-header.component.ts (hunk 2 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/components/attribute/header/configurator-attribute-header.component.ts:158]: Done!

File: feature-libs/product-configurator/rulebased/components/service/configurator-storefront-utils.service.ts (hunk 5 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/components/service/configurator-storefront-utils.service.ts:146]: Done!

File: feature-libs/product-configurator/rulebased/components/attribute/header/configurator-attribute-header.component.ts (hunk 4 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/components/attribute/header/configurator-attribute-header.component.ts:177]: Done!

File: feature-libs/product-configurator/rulebased/core/facade/configurator-groups.service.ts (hunk 6 above)
Comment (Larisa-Staroverova) [reply feature-libs/product-configurator/rulebased/core/facade/configurator-groups.service.ts]: Done

**Review by ChristophHi — APPROVED** [review]

**Pull request merged**